File: software/examples/ruby/example_button_callback.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_lcd_16x2'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change XYZ to the UID of your LCD 16x2 Bricklet

ipcon = IPConnection.new # Create IP connection
lcd = BrickletLCD16x2.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register button pressed callback
lcd.register_callback(BrickletLCD16x2::CALLBACK_BUTTON_PRESSED) do |button|
  puts "Button Pressed: #{button}"
end

# Register button released callback
lcd.register_callback(BrickletLCD16x2::CALLBACK_BUTTON_RELEASED) do |button|
  puts "Button Released: #{button}"
end

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/javascript/ExampleButtonCallback.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'XYZ'; // Change XYZ to the UID of your LCD 16x2 Bricklet

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var lcd = new Tinkerforge.BrickletLCD16x2(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function (error) {
        console.log('Error: ' + error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

// Register button pressed callback
lcd.on(Tinkerforge.BrickletLCD16x2.CALLBACK_BUTTON_PRESSED,
    // Callback function for button pressed callback
    function (button) {
        console.log('Button Pressed: ' + button);
    }
);

// Register button released callback
lcd.on(Tinkerforge.BrickletLCD16x2.CALLBACK_BUTTON_RELEASED,
    // Callback function for button released callback
    function (button) {
        console.log('Button Released: ' + button);
    }
);

console.log('Press key to exit');
process.stdin.on('data',
    function (data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
